### PR TITLE
Adding Drain, corresponding brush and new flag

### PR DESCRIPTION
**Pull request by Firigion**
Drain will remove all liquid connected to the block the player is standing in. By default it does it with some medium range (I think 25?), but the user can specify another value. If there's a selection and the player is standing within it and doesn't specify a radius, then the drain is performed within the selection.

The drain brush is basically the same, but it begins draining in the block the user clicked. Since brush traces usually go through liquids to target the blocks behind them, I added a new flag `-l` that when applied to the creation of a brush, will make that brush target liquids as well as blocks. Drain brush defaults to that behavior, of course. While I was adding the behavior for the flag, I also fixed one of the bugs in #28, now there's only one missing.

The implementation of drain is using the flood fill functionality in the background. Maybe there's a better way to do it, but that's what I could come up with.

Also fixed an unreported bug in `selection move` caused by it using an old function that had been deleted.

### CODE REVIEW

**Review by BisUmTo — DISMISSED** [review]

**Comment by Firigion** [comment]
This PR now also contains #55, since it was merged into this one. Fo read that one if you want to comment on the functionality added there.

File: world-edit.sc [hunk 1]
```diff
@@ -1207,6 +1277,19 @@ global_brush_shapes={
             if(block(start)==block, return());
             _flood_generic(block, axis, start, _(pos, outer(start), outer(radius)) -> _euclidean_sq(pos, start) <= radius*radius,  flags);
         ),
+
+
+    'flood'->_(pos, args, flags) -> (
+            [block, radius, axis] = args;
+            _flood_fill(block, pos, axis, radius, flags);
+            add_to_history('action_flood', player())
+        ),
```
Comment (Ghoulboy78) [inline world-edit.sc:1286]: 2 entries for 'flood', which one do you wanna keep?

File: world-edit.sc [hunk 2]
```diff
@@ -1831,6 +1910,107 @@ set_in_selection(block,replacement,flags)->
...
+    if( !no_selection && radius==null && _is_inside_selection(start), 
+        _flood_tester(pos)->_is_inside_selection(pos),
+        if(radius==null, radius=25);
+        _flood_tester(pos, outer(start), outer(radius)) -> _sq_distance(pos, start) <= radius*radius
+    );
+
+    _flood_generic(block, axis, start,flags);
+);
+
+_sq_distance(p1, p2) -> reduce(p1-p2, _a + _*_, 0);
```
Comment (Ghoulboy78) [inline world-edit.sc:1955]: Why not use _euclidean_sq function? It's already imported and used throughout the app...

**Review by Ghoulboy78 — CHANGES_REQUESTED** [review]
Pretty good, just some things to worked out.

File: world-edit.sc (hunk 2 above)
Comment (Firigion) [reply world-edit.sc]: Because it wasn't when I first wrote it, then I forgot to change it...

File: world-edit.sc (hunk 1 above)
Comment (Firigion) [reply world-edit.sc]: Woops, I thought I had deleted the old one!

**Review by Ghoulboy78 — APPROVED** [review]

**Pull request merged**